articles/logging-with-winston/index.md

```diff
@@ -0,0 +1,548 @@
+
+- It is simple to use and configurable. Winston is simple to configure with your application.
+- Log [levels](https://github.com/winstonjs/winston#logging-levels) (priorities). Winston provides logging levels. They indicate the log priority; this gives you the ability to sort out critical logs from logs requiring less attention. For example: `{error: 0, warn: 1, info: 2, verbose: 3, debug: 4, silly: 5}`. In this case, an `error` log is given priority over a `verbose` log.
+- Logging [channels](https://github.com/winstonjs/winston#transports) (transports). A good logger has a different way of choosing your log output destination. With Winston, you can send and save logs in different ways, such as files, databases, emails, and console.
+- Log [formats](https://github.com/winstonjs/winston#formats). Winston provides you with several log formats. For example, when saving a log to a mongo database, the log format needs to be in JSON format. Winston solves that for you.
+- Log [profiling](https://github.com/winstonjs/winston#profiling). Winston helps you profile a block of code and measure the time the code takes to be successfully executed.
+
+### Using Winston
+
+To get started, make sure you initialize a Node.js project using `npm init`. Here's a [guide](/engineering-education/beginner-guide-to-npm/) that will help you understand NPM in detail.
+
+Install the Winston package using the command `npm install winston`.
+
+Logging with Winston is simple, with just four steps, as shown in the example below, and you have your log recorded.
+
+- Add the Winston module with a `require()` function.
+
+```js
+const winston = require('winston');
+```
+- Creating Winston configuration object. Winston requires at least one transport to create a log. A transport is where the log is saved.
+
+```js
+const logConfiguration = {
+    'transports': [
+        new winston.transports.Console()
+    ]
+};
+```
+- Creating a logger and pass it to the Winston configuration object.
+
+```js
+const logger = winston.createLogger(logConfiguration);
+```
+- Log your message.
+
+```js
+// Log a message
+logger.log({
+// Message to be logged
+    message: 'Hello, Winston!',
+
+// Level of the message logging
+    level: 'info'
+});
+// Log a message
+logger.info('Hello, Winston!');
+```
+
+The above example shows you how to log to the console.
+
+```js
+{"message":"Hello, Winston logger a warning!","level":"warn"}
+{"message":"Hello, Winston logged an error!","level":"error"}
+```
+
+### Winston Transporters
+One of the properties of Winston is that it supports various transports such as uch as file transport. This will save a generated log message to a log file. The file is specified within your system. If the application creates its first log instance, the file will automatically be generated. Afterwards, any log will be saved to the created file.
```

Comment:
**Typo**: Remove uch as before file transport.

Reply:
@kimkimani, can you fix this?

Reply:
> @kimkimani, can you fix this?

yeah @louisefindlay23, pushed on [e54500c](https://github.com/section-io/engineering-education/pull/632/commits/e54500c0046b419a86181d144bbc8a9a4307bcf6)